dotnet/samples/Concepts/FunctionCalling/FunctionCalling.cs

```diff
@@ -0,0 +1,459 @@
+///    which means all kernel functions are provided or advertised to the AI model. If a list of functions is provided,
+///    only those functions are advertised to the AI model. An empty list means no functions are provided to the AI model,
+///    which is equivalent to disabling function calling.
+///
+/// ** Function calling behavior **
+///    The <see cref="FunctionChoiceBehavior.Auto"/> behavior allows the model to decide whether to call the functions and, if so, which ones to call.
+///    The <see cref="FunctionChoiceBehavior.Required"/> behavior forces the model to call the provided functions. The behavior advertises functions in the first
+///    request to the AI model only and stops advertising them in subsequent requests to prevent an infinite loop where the model keeps calling functions repeatedly.
+///    The <see cref="FunctionChoiceBehavior.None"/> behavior tells the AI model to use the provided functions without calling them to generate a response.
+///    This behavior is useful for dry runs when you want to see which functions the model would call without actually invoking them.
+///
+/// ** Function invocation **
+///    The <see cref="FunctionChoiceBehavior.Auto"/> and <see cref="FunctionChoiceBehavior.Required"/> supports two modes of function invocation: manual and automatic:
+///    * Automatic function invocation mode enables invocation of all functions called by AI model automatically by SK.
+///      The results of these function invocations are added to the chat history and sent to the model automatically in the following request.
+///      The model then reasons about the chat history and then calls functions again or generates the final response.
+///      This approach is fully automated and requires no manual intervention from the caller. The automatic invocation mode is enabled by default.
+///    * Manual invocation mode returns all function calls requested by the AI model to the AI API caller. The caller is fully responsible
+///      for the invocation phase where they may decide which function to call, how to handle exceptions, call them in parallel or sequentially, etc.
+///      The caller then adds the function results/exceptions to the chat history and returns it to the model, which reasons about it
+///      and then calls functions again or generates the final response. This invocation mode provides more control over the function invocation phase to the caller.
+///      To enable manual invocation, the caller needs to set the `autoInvoke` parameter to `false` when specifying either <see cref="FunctionChoiceBehavior.Auto"/>
+///      or <see cref="FunctionChoiceBehavior.Required"/> in the <see cref="PromptExecutionSettings"/>.
+///
+///    SK supports only sequential invocation of functions in the automatic invocation mode at the moment. The concurrent invocation mode is only possible
+///    in the manual invocation mode.
```

Comment:
`concurrent invocation mode` makes it sound like a there is a feature for this.
Consider instead `To invoke functions concurrently, a caller will need to do this manually.`